viewing_party/party.py

```diff
@@ -1,23 +1,132 @@
+def get_watched_avg_rating(user_data):
+    rating_sum = 0
+    if len(user_data["watched"]) < 1:
+        return 0.0
+    for movie in user_data["watched"]:
+        rating_sum += movie["rating"]
+        rating_sum = float(rating_sum)
+    avg_rating = rating_sum / len(user_data["watched"])
+    return avg_rating
 
 
+def get_most_watched_genre(user_data):
+    genre_list = []
+    if len(user_data["watched"]) < 1:
+        return None
+    for movie in user_data["watched"]:
+        genre = movie.get("genre")
+        genre_list.append(genre)
+    genre_count = max(set(genre_list), key=genre_list.count)
+    return genre_count
+
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+
+def get_unique_watched(user_data):
+    my_unique_movies = []
+    for my_movies in user_data["watched"]:
+        unique_movie = True
+        for friend in user_data["friends"]:
+            for friends_movies in friend["watched"]:
+                if my_movies["title"] == friends_movies["title"]:
+                    unique_movie = False
+        if unique_movie is True:
+            my_unique_movies.append(my_movies)
+    return my_unique_movies
+
+
+def get_friends_unique_watched(user_data):
+    friends_unique_movies = []
+    for friend in user_data["friends"]:
+        for friends_movies in friend["watched"]:
+            unique_movie = True
+            for my_movies in user_data["watched"]:
+                if friends_movies["title"] == my_movies["title"]:
+                    unique_movie = False
+            if unique_movie is True:
+                movie_already_in_list = False
+                for movie in friends_unique_movies:
+                    if friends_movies["title"] == movie["title"]:
+                        movie_already_in_list = True
+                if movie_already_in_list is False:
+                    friends_unique_movies.append(friends_movies)
+    return friends_unique_movies
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    subscriptions = user_data["subscriptions"]
+    user_watched = user_data["watched"]
+    friends = user_data["friends"]
+    recommended_movies = []
+    user_watched_titles = [movie["title"] for movie in user_watched]
+    for friend in friends:
+        for movie in friend["watched"]:
+            if movie["title"] not in user_watched_titles and movie["host"] in subscriptions:
+                if movie not in recommended_movies:    
+                    recommended_movies.append(movie)
+    return recommended_movies
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+
+def get_new_rec_by_genre(user_data):
+    most_watched_genre = get_most_watched_genre(user_data)
+    print(most_watched_genre)

```

Comment:
Remember to remove print statements that you use for debugging before submitting projects!